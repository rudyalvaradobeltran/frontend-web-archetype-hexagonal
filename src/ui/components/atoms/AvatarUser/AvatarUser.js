import React from 'react';
import PropTypes from 'prop-types';

import StyledAvatar from './styles';

const AvatarUser = ({ name, phone }) => (
  <StyledAvatar>
    <img src="https://miro.medium.com/max/360/1*W35QUSvGpcLuxPo3SRTH4w.png" alt="avatar_img " />
    <p>{name}</p>
    <p>{phone}</p>
  </StyledAvatar>
);

AvatarUser.propTypes = {
  name: PropTypes.string,
  phone: PropTypes.string,
};
AvatarUser.defaultProps = { name: '', phone: '' };

export default AvatarUser;
