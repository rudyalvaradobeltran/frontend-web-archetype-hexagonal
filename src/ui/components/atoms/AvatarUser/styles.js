import styled from 'styled-components';

const StyledAvatar = styled.div`
  border: 2px solid gray;
  text-align: center;

  & img {
    width: 100%;
  }
`;

export default StyledAvatar;
