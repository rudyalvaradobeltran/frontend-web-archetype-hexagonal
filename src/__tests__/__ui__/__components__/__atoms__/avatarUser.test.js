import React from 'react';
import { create } from 'react-test-renderer';

// Mocks
import MockProvider from '../../../../__mocks__/mockProvider';

// Components
import AvatarUser from '../../../../ui/components/atoms/AvatarUser/AvatarUser';

describe('AVATAR USER', () => {
  const component = (children) => <MockProvider>{children}</MockProvider>;

  it('Should render correctly (snapshot) with default props', () => {
    const wrapper = create(component(<AvatarUser />));
    expect(wrapper.toJSON()).toMatchSnapshot();
  });

  it('Should renders correctly (snapshot) whit  props', () => {
    const wrapper = create(component(<AvatarUser name="Jhon Doe" phone="87745448" />));

    expect(wrapper.toJSON()).toMatchSnapshot();
  });
});
