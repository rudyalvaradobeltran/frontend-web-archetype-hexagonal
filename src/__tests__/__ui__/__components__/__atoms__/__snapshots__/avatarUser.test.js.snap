// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AVATAR USER Should render correctly (snapshot) with default props 1`] = `
<div
  className="sc-bdvvaa bBZNia"
>
  <img
    alt="avatar_img "
    src="https://miro.medium.com/max/360/1*W35QUSvGpcLuxPo3SRTH4w.png"
  />
  <p>
    
  </p>
  <p>
    
  </p>
</div>
`;

exports[`AVATAR USER Should renders correctly (snapshot) whit  props 1`] = `
<div
  className="sc-bdvvaa bBZNia"
>
  <img
    alt="avatar_img "
    src="https://miro.medium.com/max/360/1*W35QUSvGpcLuxPo3SRTH4w.png"
  />
  <p>
    Jhon Doe
  </p>
  <p>
    87745448
  </p>
</div>
`;
